Automattic/wp-calypso
# Gutenboarding: Send "guessed" browser timezone string for new sites

#### Changes proposed in this Pull Request

The site creation used in production now creates sites with a best guess of what their timezone should be (#38891). This does the same thing, but in gutenboarding.

* Send `timezone_string` param when creating new site in gutenboarding
* Guess timezone using code from `lib/i18n-utils/guess-timezone`
* Remove `moment-timezone` dependency from `guessTimezone()` so we don't have to include it in gutenboarding
* Update existing uses of `guessTimezone()` so they deal with possible failure since `Intl` API may not be available (it is in all the browsers we support, but we still want the page to mostly work on unsupported browsers). If `Intl` is unavailable:
  * when creating a site, then the `timezone_string` param isn't sent
  * when viewing a site's general settings page then we don't suggest a timezone
  * when booking a concierge session then the form defaults to UTC

This PR pulls in non-packaged code from Calypso which isn't ideal, given we're trying to keep Calypso dependencies out of gutenboarding. However `i18n-utils` is imported in a few places already, so I think it'd be good to come up with a generic solution that fixes all of these imports at once.

#### Testing instructions



* Create a new site via `/gutenboarding`
* Check the API request to `/sites/new` contains the correct `timezone_string` param
* The site's time zone should be that of the computer used to create the site (visible at `/settings/general/{site_slug}`)

Testing other uses of `guessTimezone()` haven't broken:

* Create a new site starting from `/start` and check the site's timezone is correct
* The timezone suggestion button at `/settings/general/{siteslug}` should still work
* The timezone option at the conceirge signup form should still default to the guessed timezone `/me/concierge/{siteslug}/book`

Can't think of a way to test what happens when `Intl` isn't available in a browser other than hardcode `guessTimezone()` to always return `undefined`.
